pkg/types/utils.go

```diff
@@ -45,3 +48,25 @@ func identifyAndGetInstance(resName string) (runtime.Object, error) {
 	}
 	return nil, errors.Errorf("failed to identify instance for: %s", resName)
 }
+
+func CastToBindingType[T any](dataValue *data.StructValue, destBindingType bindings.BindingType) (T, error) {
+	converter := bindings.NewTypeConverter()
+	obj, errs := converter.ConvertToGolang(dataValue, destBindingType)
+	if len(errs) > 0 {
+		t, _ := obj.(T)
```

Comment:
Is there any need for assigning `t` here? I think you can probably just return `nil, errs[0]`

Reply:
can't return nil as per the compiler

Reply:
but I found a better way to do this